src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -414,62 +415,90 @@ internal static bool IsModuleMatchingModuleSpec(PSModuleInfo moduleInfo, ModuleS
 
         internal static Version GetManifestModuleVersion(string manifestPath)
         {
-            if (manifestPath != null &&
-                manifestPath.EndsWith(StringLiterals.PowerShellDataFileExtension, StringComparison.OrdinalIgnoreCase))
+            try
             {
-                try
-                {
-                    var dataFileSetting =
-                        PsUtils.GetModuleManifestProperties(
-                            manifestPath,
-                            PsUtils.ManifestModuleVersionPropertyName);
+                Hashtable dataFileSetting =
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestModuleVersionPropertyName);
 
-                    var versionValue = dataFileSetting["ModuleVersion"];
-                    if (versionValue != null)
+                object versionValue = dataFileSetting["ModuleVersion"];
+                if (versionValue != null)
+                {
+                    Version moduleVersion;
+                    if (LanguagePrimitives.TryConvertTo(versionValue, out moduleVersion))
                     {
-                        Version moduleVersion;
-                        if (LanguagePrimitives.TryConvertTo(versionValue, out moduleVersion))
-                        {
-                            return moduleVersion;
-                        }
+                        return moduleVersion;
                     }
                 }
-                catch (PSInvalidOperationException)
-                {
-                }
             }
+            catch (PSInvalidOperationException) { }
 
             return new Version(0, 0);
         }
 
         internal static Guid GetManifestGuid(string manifestPath)
         {
-            if (manifestPath != null &&
-                manifestPath.EndsWith(StringLiterals.PowerShellDataFileExtension, StringComparison.OrdinalIgnoreCase))
+            try
             {
-                try
+                Hashtable dataFileSetting =
+                    PsUtils.GetModuleManifestProperties(
+                        manifestPath,
+                        PsUtils.ManifestGuidPropertyName);
+
+                object guidValue = dataFileSetting["GUID"];
```

Comment:
I assume there's no static catalogue of module manifest keys, so that rather than `dataFileSetting["GUID"]` we could do `dataFileSetting[ModuleManifestKeys.GUID]`?

Reply:
I found the static type `ConfigFileConstants` in `InitialSessionStateProvider.cs` where a const string `Guid = "GUID"` is defined, but that type is not suitable to be used here because it's for remoting configuration purpose.
We can add const strings for all manifest keys in a separate PR.

Reply:
There seem to be some strings in PSUtils.cs:
https://github.com/PowerShell/PowerShell/blob/4fc784e840b8df00d2552045c6ed52d027659d2b/src/System.Management.Automation/utils/PsUtils.cs#L477-L479

Are they one place we store these?